pkg/availability/multisigcollector/internal/parts/batchreconstruction/batchreconstruction.go

```diff
@@ -132,9 +134,17 @@ func IncludeBatchReconstruction(
 			return nil
 		}
 
-		if _, ok := state.RequestState[reqID]; !ok {
+		if req, ok := state.RequestState[reqID]; !ok {
 			// Ignore a message with an invalid or outdated request id.
 			return nil
+		} else if _, ok = req.Txs[batchID]; !ok {
+			// Ignore a message with a batchID that was not requested.
+			return nil
+		}
+
+		if txs == nil || sliceutil.Contains(txs, nil) {
+			// Ignore an empty txs list or a txs list with a nil tx.
+			return nil
 		}
```

Comment:
I would log a warning on each of these occasions, saying that we ignore the message and why.